### ISSUE TITLE
[Danger] [Important] Asyncronous writing can destroy URPMS repos

### ISSUE THREAD

**Issue by paulcarroty** [UnitedRPMs/packages#112]
We need the one worker for collecting new packages *right now*. Because I don't want to lost all. Sshd and SF support parallel connections, so we can very easy shout our legs.

Requirements:
1) open license and sources - the URPMS style from roots
2) readable code
3) no black boxes like private containers, Dockerfiles lack and other shit

I now use the https://github.com/paulcarroty/travis-rpm-repobuilder, we should improve it improve @kuboosoft tools, don't matter. 

Since this moment I suggest to **freeze all pushes** to our binary repositories.

**Comment by kuboosoft** [comment]
1). All my code has [MIT licences](https://fedoraproject.org/wiki/Licensing:Main?rd=Licensing)... then we need check your licences... and your various git repositories (is confuse); I do not understand why you have so many repositories with the same tools

**My tools** 2

**Sync SF tool**
[sync-unibuild](https://github.com/kuboosoft/united-build/blob/master/sync-united)

- Makes a backup of all SF
- Checks and Downloads new packages in the cloud
- Syncs all a make a createrepo

**Build tool**
[unibuild/urpms](https://github.com/kuboosoft/united-build/blob/master/urpms)
- It can upload to the cloud if is enabled with "-c true" if not, it works in test mode...
All results of unibuild will uploading to a cloud (You have full acces since your creation; forgot it?)

2) readable code? what? It is incoherent, code is code, buddy...
3) Black boxes? sorry but I don't want people checking ours private data... also depend of travis... with other CI service we can use private docker...
Only because travis doesn't support docker private we need accept it...? Sorry but you want close all...

Problems found in your code; how to find the real package, is a build for test or a finished  package? I can integrate your https://api.github.com/repos/UnitedRPMs/travis-rpm-builder/releases but if you put wrong file here; then is your problem (Theoretically the package has been tested; To synchronize.). Also I can't upload a rpm with bad release, example a release with F26 in a F25... I have filters because I found, uploaded rpms with bad release... who is affected for it? our users...

In my side I check if you are syncing; with repoview, sf and travis. You can track me in repoview, sf and codefresh... I can put a automatic announcement in slack in each sync, actually only my user receive it, in this moment.

I think all our tools are good, and each suits your tastes and needs.  I can integrate your rpms, Why did not you request it before?

**Comment by paulcarroty** [comment]
> 1). All my code has MIT licences

I use only GPLv3. But MIT is also OK.

> it is incoherent, code is code, buddy...

WTF? http://ideone.com/I0HTec http://ideone.com/O1LATD

I can't lost the hours to understand any uncommented bullshit. Sync gear must me small, compact and run fast.

I also prefer running by cron and rebuild every night.

> Black boxes? sorry but I don't want people checking ours private data

Sorry, our project was started as open source. And will finished also as open source. There's nothing to discuss because it's only major our advantage. 

* https://docs.travis-ci.com/user/environment-variables/
* https://docs.travis-ci.com/user/encrypting-files/
* https://github.com/circleci/encrypted-files

> Only because travis doesn't support docker private we need accept it...?

Only because security trough obscurity is bad. And remember, there's open source project.

**Comment by paulcarroty** [comment]
> how to find the real package, is a build for test or a finished package?

I use the next algorithm: maintainer build the packages, test it under the same day and do nothing. At every night the sync gear grab all, sort it and push to repos. It's brilliant.

You can suggest something better? Wellcome.

**Comment by paulcarroty** [comment]
> You can track me in repoview, sf and codefresh... I can put a automatic announcement in slack in each sync, actually only my user receive it, in this moment.

Announcing, waiting, locks and other it's not good in this sense. It must be the only one point to brake, otherwise changing API or service **will brake all**. It's like a blender - make a good salad or burn your house.

> Why did not you request it before?

I now write new frontend to mock-builder and compete simulate the whole project design.

**Comment by paulcarroty** [comment]
* sync-united - 300 lines of code (!)
* travis-rpm-repobuilder - 64 lines of code (LOLLLLLLL)

I now it's extremely clear what we need to use for **fast** syncing. Unless I can't see some hidden superb features into sync-united.

**Comment by kuboosoft** [comment]
> I can't lost the hours to understand any uncommented bullshit. Sync gear must be small, compact and run fast.

Me too!

> I also prefer running by cron and rebuild every night. 


With travis? The minor scheduler is each week.... 

> Sorry, our project was started as open source. And will finished also as open source. There's nothing to discuss because it's only major our advantage.

> Only because security trough obscurity is bad. And remember, there's open source project. Infrastructure should also be max open: open encryption tools and algorithms.


When you use 'COPR', what did you need to enter? Was their authentication public? You need to show that kind of data to the public; you do it; I do not... Do you want put at  risk, the safety of the project ... I don't like see travis administering the keys (inclusive encripted) and downloadig the tools directly from git... and lost time downloading depencencies... Please do not confuse security with free software.

`I use the next algorithm: maintainer build the packages, test it under the same day and do nothing. At every night the sync gear grab all, sort it and push to repos. It's brilliant.`

My "algorithm" 
1) test in my local machine; edit spec files, make a patch if is necessary.... make a src.rpm an rebuild...
2) sync with github
3) Build with unibuild,  finished the build is installed the package... we can see if it is working or not.
4) Sync when the cloud has various packages... If we synchronize continuously our official repository; We will have problems in our users cache.

I can see various white spaces in your algorithm...

> If I need to fix something right now - just use the trigger.

I only need make a rebuild in codefresh... or a [git commit](https://github.com/kuboosoft/usync) (20 minutes with all repositories F24-F27)...

> NO way. I use a filter to detect the release for every package.
Is a joke? https://github.com/UnitedRPMs/packages/issues/110#issue-224640624
```
Skipping packages with broken dependencies:
 libtxc_dxtn           i686   1:1.0.1-1.gitef072983.fc26       unitedrpms  18 k
 libva-intel-driver    i686   1.7.2-1.fc26                     unitedrpms 377 k
```

> Announcing, waiting, locks and other it's not good in this sense and lost our time. It must be the only one point to brake, otherwise changing API or service will brake all. It's like a blender - make a good salad or burn your house.

Same risk when using any free service ...

> I now write new frontend to mock-builder and complete simulate the whole project design.


Do not get mad. You are a inestable man hahaha; and now with your fork in "Go", the developer abandoned it and now you do the same. Remember my words when I told you, "If you know Golang, welcome, you can work with "travis-mock" and me with unibuild tool; You  make a tool, with the knowledge you have. "is unusable for me"."

And I repeat again the same words... also I need a intelligent tool and a cloud for upload and make backups for any disaster...

**Comment by kuboosoft** [comment]
@paulcarroty if you change my answer, then is a serious problem... becase when a people put a real problem you avoid it... unibuild and united-sync make me the easy life... your tools not (With the exception of,  your mirror synchronization tool...) you compare my sync tool based in 360 lines is fool; but is best designed i in comparation to your basic tool; various functions is disabled in united-sync; I can delete it and reduce to 30 lines...; but the tools is changing constantly and is adapted to the needs...

`I completely understand the design and code of mock-travis, and also added many cool features to my fork. I can rewrite it right now for Node.js and make build from browser, but 'rewrite for rewrite' is stupid. Backend should do one thing - build a package and build it fast. Mock-travis can do it.
`
I always had reason...

We now have backups into Gitlab, there's not a problem anymore. If you need an something new - let's create it. We should create rock-solid API for critical points: repos, reposync, mirrors.

Git lab is a chronicle of an announced death; They will reduce its quota, I read the news is official, and I shared it about 2 weeks in your Slack user....

> We don't need backups inside repobuilder, it makes him slowly. If you need a third backup - modify the mirrorsync with your code.

I have a cloud...  and my unibuild and united-sync make the task...

> Maintainer need to upload the ready packages to any hosting with stable direct links. Mega, GH, GL, don't matter. And the happy ending, automation make the rest. No locks, no checks what others do.

unibuild and united-syns are ready, a long time ago ... You can use your sync tool , but I don't can use it; reason, is unusable for my needs... put it and make a cron with it... no  matter. Our tools works fine...

**Comment by paulcarroty** [comment]
> if you change my answer, then is a serious problem... because when a people put a real problem you avoid it.
what?

>  you compare my sync tool based in 360 lines is fool; but is best designed in comparation to your basic tool; various functions is disabled in united-sync; I can delete it and reduce to 30 lines

@kuboosoft **talk is cheap, show me the code©** If you can create the faster decision - DO IT now and we make it as default immediatly. Sync should *works fast*, forget about your stupid ambitions - you make a faster tool or we select my repobuilder.

One tool - one deal: grab packages and move to repo. Finish. For backups we have the other tool and gear.

> and build with mock in containers is very slow... And if travis and closes admin privileges, your tool dead.

Add the repobuild support - deal for 10 lines of code. Next problem: VirtualBox can't be compiled withouth mock, I get only the silence insdeat of answer in Slack. Probably the same with other packages too.

> I have a cloud

I punish you and take away you cloud?

> Git lab is a chronicle of an announced death; They will reduce its quota, and I shared you it about 2 weeks in your Slack user.... also when you tool make a sync with gitlab deletes our hosted files; and if it fail? you lost all.

LOL, I just create new mirror. If one mirror die we'll can't lost all.

>  You can use your sync tool , but I don't can use it; reason, is unusable for my needs... put it and make a cron with it... no matter. Our tools works fine...

Are you blind?? You can't understand that two parallel writing can destroy the repos??

Enough for this stupid talks. If you can creare the faster tool with not more than 50lines of code. If no - sorry, someone else can do it.

**Comment by paulcarroty** [comment]
https://en.wikipedia.org/wiki/Not_invented_here

**Comment by kuboosoft** [comment]
>  talk is cheap, show me the code© If you can create the faster decision - DO IT now and we make it as default immediatly. Sync should works fast, forget about your stupid ambitions - you make a faster tool or we select my repobuilder.

The code is [here](https://github.com/kuboosoft/united-build/blob/master/sync-united), and it has a [MIT license](https://github.com/kuboosoft/united-build/blob/master/sync-united#L5)... you code not, [it is proprietary](https://github.com/paulcarroty/travis-rpm-repobuilder/blob/master/repobuilder.sh#L2)... Who is the ambitious? Who wanted to make money for the project? Is it a joke? hahaha 

We can run the two tools, a progamated sync with your tool each week; and me when i need urgent/ necessary sync with my tool... I have found  inestability of our repositories; thanks a  gitlab; and your Asyncronous stupid errors...  do you want proves? [Commit in F26](https://github.com/UnitedRPMs/f26-mirrorsync/blob/362aa44593f6a83c5cb0e94227c24833af6eca0d/mirrorsync.sh) [Commit in F27](https://github.com/UnitedRPMs/f27-mirrorsync/blob/61611f6d7f0ce7343d9b92690110db04ea014066/mirrorsync.sh#L5) ; [Commit in F24](https://github.com/UnitedRPMs/f24-mirrorsync/blob/b68e15bc3740bc1d01fee95b62b70141e5e8f2a7/mirrorsync.sh#L5)

> Enough for this stupid talks. If you can create the faster tool with not more than 50lines of code - do it. If no - sorry, someone else can do it.

My simple tool is faster including the backup in the cloud... and have various lines because it has the license (27 lines) and functions disabled... [your tool is proprietary!](https://github.com/paulcarroty/travis-rpm-repobuilder/blob/master/repobuilder.sh#L2)! and doesn't work with results in unibuild and MG...

> Are you blind?? You can't understand that two parallel writing can destroy the repos??

I understand; two parallel can destroy it; You have occasionally provoked it; And I had to solve ... I found old versions, packages rebuilt with old dependencies...

> Enough for this stupid talks. If you can create the faster tool with not more than 50lines of code - do it. If no - sorry, someone else can do it.

Why do that? Their arguments lack logic, the tool is adapted to be used with MG; Its tool not, it is unusable for me. And with license Paul Carroti (c)... I have dedicated a lot of time to the project; But if you break my balls constantly; Will be left alone. 

https://www.quora.com/How-do-I-stop-being-a-hypocrite

**Comment by paulcarroty** [comment]
> you code not, it is proprietary

Welcome to real world: https://github.com/paulcarroty/travis-rpm-repobuilder/blob/master/LICENCE

> We can run the two tools

No. Live with a bomb not for me.

> your tool is proprietary!!

Are you crazy?? GPLv3 is proprietary??

> My simple tool is faster including the backup in the cloud

Let's define the simple word: [sync-united](https://github.com/kuboosoft/united-build/blob/master/sync-united) - 319 lines; [urpms](https://github.com/kuboosoft/united-build/blob/master/urpms) - 972 lines, my [repobuilder](https://github.com/paulcarroty/travis-rpm-repobuilder/raw/master/repobuilder.sh) - 64 lines of code!!!

@sergiomb2 we need your vote as project member right now because the discussion lost their common sense.

**Comment by kuboosoft** [comment]
We can run the two tools, a progamated sync with your tool each week; and me when i need urgent/ necessary sync with my tool... I have found inestability of our repositories; thanks a gitlab; and your Asyncronous stupid errors... do you want proves? Commit in [F26 Commit](https://github.com/UnitedRPMs/f26-mirrorsync/blob/362aa44593f6a83c5cb0e94227c24833af6eca0d/.travis.yml#L11) in [F27](https://github.com/UnitedRPMs/f27-mirrorsync/blob/61611f6d7f0ce7343d9b92690110db04ea014066/.travis.yml#L9) ; Commit in [F24](https://github.com/UnitedRPMs/f24-mirrorsync/blob/b68e15bc3740bc1d01fee95b62b70141e5e8f2a7/.travis.yml#L9)

Other, a mirror updated each week is a problem... a mirror should run each day...  use as a backup gitlag is bad idea, reason, it is deleted each sync ( week). 

**We argued for insignificant shit; We can use both tools ...** 

1) Your tool with weekly synchronization (programmable); I can't see nothing special here... only a encripted key...

2) My tool for synchronizations without waiting, I use MG; The cloud always receives results, and every result is decharged. I do not work directly with the official repository; the tools syncs in brute when it is ready...  It's more logical

![how to work my sync tool?](https://i.imgur.com/SraG3R5.png)

**Your sync  tool works directly with the repository, you can easily break all repositories ....**

**Comment by paulcarroty** [comment]
> We can use both tools ...

Second explanation for idiot's here: if rsync from two different sources write into origin mirror in one moment then will brake it; it's like to make stuffing.

> I have found inestability of our repositories; thanks a gitlab

Bullshit.

> a mirror updated each week is a problem

Bullshit. I'm already talk why use weekly sync.

> Your sync tool works directly with the repository, you can easily break all repositories

Your hand works directly with your body, you can easily shot yourself.

So, I now understand last doings of @kuboosoft as danger to URPMS and he's **temporary deleted as project admin on Sourceforge**. I'm waiting for a @sergiomb2's vote.

**Comment by sergiomb2** [comment]
absent :smile: , I haven't follow this thread, don't like the language  you should coordinate your work , I tend to support @kuboosoft , but now I don't know what is the problem ...

**Comment by paulcarroty** [comment]
OK, I'm done, bye.

**Comment by sergiomb2** [comment]
but this time I can vote on you, Paul , I just don't read the thread  .

**Comment by kuboosoft** [comment]
@sergiomb2 simple, @paulcarroty want use your tool for sync (bad tool as my tool)... but his tool is unusable for me, for new maintainers... he want depend directly with Travis CI (I can't support it), because it has a limited time of building (60 minutes). 

- "His build tool" builds packages and upload individuas results in github; What if we have quota problems?

-  His sync tool upload directly to SF (individual rpms, without test), example: ffmpeg; if the tool uploads it, without build others packages, then our users will cry because they can't install ffmpeg package, because other packages needs the previous ffmpeg package... the people make fools task as "try to add '--allowerasing' to command line to replace conflicting packages" results? various dependencies broken, inclusive a people can remove critial parts in gnome...

- His sync tool exclude results of [MG](https://mega.nz/); then my time; updating, patching... It is useless..
- If I edit the code of @paulcarroty he will cry because don't  like MG, inclusive if it has a open source license he will delete it... but MG (the cloud) give us 50GB free, it is beatiful; our results is saved as wallet...  and our tools can works as the graphical above, and I added him to group.. . In begin I shared the pass of MG with Paul (he has full acces, since bigin); but he closes his mind... 

- His build parameter (human task) with your travis-mock is wait 1, 2... hours each build; for me it is a problem, and dangerous, because he is using 'mirror list' in [mock config](https://fedoraproject.org/wiki/Using_Mock_to_test_package_builds#Edit_.2Fetc.2Fmock.2F.2A.cfg_to_not_use_mirror_lists); I found a problem with it, with a no sincronized mirror and a package built with a old dependency; I don't have time for it...  I  am not using mirror list since this problem...

- My build tool: builds (detect massive rebuild), makes a automatic incremental release en massive rebuild; checks if exist packages already built in MG... finished a build, my build tool installs it. and finally unibuild uploads to MG the results (beatiful in mass rebuild). We can check if exist problems with our package.

My 'sync tool' downloads results of my build tool hosted in MG, and finally makes a incremental backup in MG of SF.. simple words it Interacts with MG.  Also It uses a private docker (here saves security data). I told him, I can integrate your results  of git, but he closes your mind...

**For me it is complete nonsense to fight for our tools; Both are adapted to different needs. But with a common purpose, build and synchronize.** Is not there freedom to use the tools we want?

So far his only infrastructure is git + travis; travis works fine with its cron tasks...

> So, I now understand last doings of @kuboosoft as danger to URPMS and he's temporary deleted as project admin on Sourceforge until the @sergiomb2's vote resolve the situation.

wtf? Maybe @paulcarroty  want to close the project, or take full control of the project, maybe he wants to work alone; I do not know his dark interests (Unitedrpms is formed by altruistic contribution; A hidden society.). I can stop right now my contribution, and focus me on another project; @sergiomb2  Do you understand now why I did not want to mix postinstallerf to unitedrpms?

**Comment by sergiomb2** [comment]
TLDR, I just read the later paragraphs , well , SF access is not exactly a big thing , 2nd to use SF we just could have open software , and some non redistributable are questionable , but we can see as private use ( if you understand what I mean) , so you should grow up :) , and I 'm admin ! of unitedrpms !  
so I don't understand , can you make an agreement with what is the way of sync repos ? , why don't make 2 repos ? I like the image explanation of @kuboosoft  :smile: .
@paulcarroty 2 resyncs break the things, I understand  , but you have to make an agreement how it do the resyncs , the problem it self is TLDR. REALLY don't understand you ,  @kuboosoft  could you publish on other directory , and @paulcarroty sync to master with his script ?

**Comment by kuboosoft** [comment]
@sergiomb2 

> I like the image explanation of @kuboosoft
I am a teacher hahaha; then I need explain it with easy way :)

I use MEGA as my directory in the cloud; isn't perfect but 50GB free is beatiful; I can give you acces to my MEGA Account, also we can share, make a group of job, because this capabilities is enabled in MEGA. I suggested to Paul, use his tool running with travis and its included cron jobs, each week (because isn't have a cron daily). also I add now, he can integrate my updates in MG to the script...

Also I can integrate the results of mock-travis to my resync script; In the past this was my idea; But this man has several repositories with the same tool; And I did not find the actual results; Then I stopped it. But now Paul publishes the results of mock-travis ... is great but, everything is left to the last minute ....
To publish in another directory the results of unibuild, implies a double workload. And a longer construction time (bad for travis and other services). Imagine; If you check all changelog, you can find who is updating, patching, testing our packages :( ... At this moment there is no way to build mythtv; For Fedora 26 and 27, I do not have time for it (And I have the infrastructure for it). Ask Paul how to build mythtv with Travis or better with circle ? (It needs 3 hours 20 minutes); or build chromium-freeworld with ([gitlab](https://github.com/docker/hub-feedback/issues/727) has problems with extracting files in %prep) 

This coupled with the development of unibuild (In the future I will migrate it to python) ... Why I need to run my synchronization tool? Simple, mitigation, quickly solve problems, packet tests, sychronization with Mega (Which will disappear from now on.).

So far I have not broken anything; However he has caused it. And I have mitigated everything; Slack is evidence of my claims ...

I am not upset (I can regain my access to SF); Because I am happy of the hand offered; And see that the project is useful for many. And I know, many love the project, and others do not. See you soon.

If users are interested in maintaining unitedrpms repositories, they are welcome. See you soon. :)

**Comment by sergiomb2** [comment]
TLTR (too long to read) 

> To publish in another directory the results of unibuild, implies a double workload. And a longer construction time (bad for travis and other services), and it maintains a complete backup.

@paulcarroty solutions ? , don't bring me problems, bring me solutions .

**Comment by sergiomb2** [comment]
@paulcarroty I going readd kuboosoft to SF ,  please communicate each other , I vote in kubo  , kubo have done an amazing work , and we should try to help him, we may talk on slack if you prefer ... 

Cheers

**Comment by sergiomb2** [comment]
I agree with [Principles-of-UnitedRPMs-Project ](https://github.com/UnitedRPMs/packages/wiki/Principles-of-UnitedRPMs-Project) Highlighting point 3 and 4

**Comment by sergiomb2** [comment]
@kuboosoft I add you back in SF, also please try integrate Paul's work , this project begins with Paul ideas, and we have create this project together so we must stay united or if we stop, we should change the name of the project ... 

Cheers.

**Comment by kuboosoft** [comment]
@sergiomb2 We can not fight like dogs and cats; With silly fights; One day I eliminate you, another eliminate me; One day someone erases the project; I have so many problems and to include more, it is not good for me, for all. I abandon the project; I contributed to the wiki so the project can live; And maintain harmony among its members. The project has been a good experience for me. I will not stop. See you soon. I wish the best for the project and its members. :smile_cat:
